src/main/java/backend/Logic.java

```diff
@@ -232,7 +232,13 @@ public void removeUnusedModels(Set<String> reposInUse) {
     }
 
     public ImmutablePair<Integer, Long> updateRemainingRate(ImmutablePair<Integer, Long> rateLimits) {
-        uiManager.updateRateLimits(rateLimits);
+        uiManager.updateRateLimits(rateLimits, true);
+        return rateLimits;
+    }
+
+    public ImmutablePair<Integer, Long> updateRemainingRateWithoutUpdateRefreshRate(ImmutablePair<Integer,
+                                                                                    Long> rateLimits) {
```

Comment:
Yes I agree, I think my previous implementation is abit improper. Updating of sync refresh rate should not be bundled together with updating of the remaining rate of the apiBox as they need to be called at different time from different location. I have created two different event for both of them. Please let me know if its ok.